app/controllers/users_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+  def index
+    users = User.all
+
+    render json: users, status: 200
+  end
+
+  def show
+    user = User.find(params[:id])
+
+    render json: user, status: 200
+  end
+
+  def create
+    user = User.new(user_params)
+
+    if user.save!
+      render json: user, status: 201
+    else
+      render json: {error: "User is invalid"}, status: 400
```

Comment:
Its tough to interpret, but [400](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) specifies malformed syntax when the server could not understand the request.  422 is typically used for when the client sends a request that was understood but not valid or appropriate.